core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -120,9 +120,11 @@ public static synchronized CoreSocketFactory getInstance() {
   @VisibleForTesting
   // Returns a listenable, scheduled executor that exits upon shutdown.
   static ListeningScheduledExecutorService getDefaultExecutor() {
-    // TODO(kvg): Figure out correct way to determine number of threads
+    // Set the number of threads to 8. This should be enough threads that most users
```

Comment:
Let's split this out into a separate PR since it's technically the fix of the bug.